### ISSUE TITLE
Question about RL

### ISSUE THREAD

**Issue by feimo-49** [Yueeeeeeee/HRPO#2]
Thanks for your wonderful work.

I'm trying to reproduce your work, but when I try to train the model with the train script "python hrpo_gsm8k.py --", I find that the reward remain 0 in the RL process, and the responses don't contain the "####" answer start trigger.

I'm appreciate for your contribution and expect your help.

**Comment by Yueeeeeeee** [comment]
Thanks for your interest in our work, we only reward the model when both the format and prediction is correct. Could you let me know what configuration are you using to run the experiments?

**Comment by feimo-49** [comment]
Thanks for your kindly response, I run the llama3.1-8b-instruct and  qwen2.5-3b-instruct models with the script provided in the README on the GSM8k dataset, i.e., group_size=8, residual_r_min=0.98; other parameter settings are default in the hrpo_gsm8k.py.

In the RL process, I observe the reward always remain 0 for llama3.1-8b-instruct and mostly remain 0 or a small value for qwen2.5-3b-instruct model ( mainly due to the format ). 

``` Batch accuracy: NNNNNNNN 
1/8 responses (answer: 13):
{reasoning_process}
To solve this problem, we need to break down the time into manageable parts and account for the intermission and the duration of the songs.

1. Total concert time: 60 minutes (1 hour)
2. Intermission time: 10 minutes
3. Remaining time for the actual songs: 60 minutes - 10 minutes = 50 minutes
4. Duration of each song, except one: 5 minutes
5. Duration of the one song that lasted 10 minutes: 10 minutes

Let's calculate the number of 5-minute songs and the 10-minute song:

1. Number of 5-minute songs:
\[ \text{Number of 5-minute songs} = \frac{50 \text{ minutes}}{5 \text{ minutes/song}} = 10 \text{ songs} \]

2. Number of 5-minute songs that can fit into the 50 minutes:
\[ \text{Number of 5-minute songs} = \frac{50 \text{ minutes}}{5 \text{ minutes/song}} = 10 \text{ songs} \]

3. Total number of songs:
\[ \text{Total number of songs} = 10 \text{ (5-minute songs)} + 1 \text{ (10-minute song)} = 11 \text{ songs} \]

The final answer is 11 songs.

{answer}
11 
==================================================
{'loss': 89.0996, 'grad_norm': 24075.25390625, 'learning_rate': 1.2834224598930483e-06, 'rewards/reward_func': 0.0, 'reward': 0.0, 'reward_std': 0.0, 'embeds_ratio': 0.9947480410337448, 'hidden_ratio': 0.10235349461436272, 'completion_length': 272.4375, 'kl': 17819.92765816301, 'epoch': 0.03}
```

```
Batch accuracy: NNNNNNNY 
1/8 responses (answer: 1600):
Let's break down the problem step by step to find the total number of words in the poem.

1. **Determine the number of lines in the poem:**
   - Each stanza has 10 lines.
   - There are 20 stanzas.
   \[
   \text{Total number of lines} = \text{Number of stanzas} \times \text{Number of lines per stanza} = 20 \times 10 = 200
   \]

2. **Determine the number of words in each line:**
   - Each line has 8 words.
   \[
   \text{Total number of words} = \text{Total number of lines} \times \text{Number of words per line} = 200 \times 8 = 1600
   \]

So, the total number of words in the poem is \(\boxed{1600}\). 
==================================================
{'loss': 17.0907, 'grad_norm': 222863.390625, 'learning_rate': 1.3636363636363636e-06, 'rewards/reward_func': 0.03125, 'reward': 0.03125, 'reward_std': 0.0883883461356163, 'embeds_ratio': 0.9947458952665329, 'hidden_ratio': 0.10237472504377365, 'completion_length': 290.625, 'kl': 3418.134404145181, 'epoch': 0.03}
```

**Comment by Yueeeeeeee** [comment]
Thank for the details, did you update anything in the codebase? As we use a GRPO-like implementation, the RL loss should usually be around 0 here. But you KL, loss and grad norm values all seem to be too large here.
